philology/src/main/java/com/jcminarro/philology/ViewTransformer.kt

```diff
@@ -1,12 +1,57 @@
 package com.jcminarro.philology
 
+import android.content.Context
 import android.util.AttributeSet
 import android.view.View
 
-interface ViewTransformer{
+interface ViewTransformer {
     fun reword(view: View, attributeSet: AttributeSet): View
 
-    fun setTextIfExists(attributeSet: AttributeSet, index: Int, setTextResAction: (Int) -> Unit){
-        attributeSet.getAttributeResourceValue(index, -1).takeIf {it != -1}?.let {setTextResAction(it)}
+    fun setTextIfExists(
```

Comment:
This method is too complex and we are breaking the public API of `ViewTransformer` that is used by consumers for its own `CustomViews`.
Instead of that, we should keep the previous `setTextIfExists()` method and provide another to assign text from style.
This new method could do the logic to check if there is any text already assigned on the xml file and then call to the previous `setTextIfExists()` method or get the value from the style

Reply:
@JcMinarro I have made the old `setTextIfExists` public to provide compatibility with public API. 

Let me know you think about it.